packages/flutter_tools/lib/src/base/bot_detector.dart

```diff
@@ -95,10 +95,16 @@ class AzureDetector {
     }
     final HttpClient client = _httpClientFactory()
       ..connectionTimeout = const Duration(seconds: 1);
+    assert(client != null);
     try {
       final HttpClientRequest request = await client.getUrl(
         Uri.parse(_serviceUrl),
       );
+      if (request == null) {
+        // The HttpClient we have is not conforming to the interface - we may be
```

Comment:
In general, I don't think we should structure application code to be defensive against bad/incomplete fakes and mocks.

If there is a problem with getting a good fake or mock plumbed to this point, then it would be more sustainable to fix that problem.

Reply:
The problem is that with tests being randomized, it's hard to tell if some test that did not used to install its own HttpClient before this check ran now runs first and makes this throw an exception.  We'd have to audit all of the places we override or fake an HttpClient, and make sure that they all implement getUrl properly. Although on CI, other checks will return true before this one has to run.

Reply:
I can split this change out since I think on CI it won't get tripped (we'll see we're on Cirrus first).

I'm also not clear on why we're counting this as a bot - someone could be using an Azure VM for development right?

Reply:
Reverting this change since it's not essential to the goal of this PR - we can catch up later about how to properly handle this.